osp/core/ontology/parser.py

```diff
@@ -338,6 +342,50 @@ def _add_reference_style_triples(self, reference_styles):
                     rdflib.Literal(True)
                 ))
 
+    def _check_duplicate_labels(self, namespace, identifier):
+        # Recycle some methods from the Namespace class. A namespace class
+        # cannot be used directly, as the namespace is being spawned.
+        placeholder = type('', (object, ),
+                           {'_iri': rdflib.URIRef(namespace),
+                            '_graph': self._graphs[identifier]})
+
+        def in_namespace(item):
+            return OntologyNamespace.__contains__(placeholder, item)
+
+        def labels_for_iri(iri):
+            return OntologyNamespace._get_labels_for_iri(placeholder, iri,
+                                                         lang=None,
+                                                         _return_literal=True)
+        # Finally check for the duplicate labels.
```

Comment:
```
labels = sorted((o, s) for s, _, o in graph.triples(None, rdflib.RDF.label, None) if s.startswith(namespace))
```
and then it should be easy to check if there are two s for the same label, isn't it?
    

```

Reply:
This function could be reduced to a couple of lines, no?

Reply:
It's a bit more complicated than that because one has to consider both RDF and SKOS labels, and when two subjects have the same RDF label, but one of the items also has an SKOS label then "there is no conflict", in the sense that the preferred labeling for such item is SKOS, etc. But I am having a look at this and trying to make the whole function simpler without duplicating code from namespace.

Reply:
Ah okay, I see. Maybe you can still make it a bit simpler, then I'd be fine with merging.

Reply:
Check the code again. I do not know if much else can be done if one wants to return all the conflicting labels in the exception, and not just the first one found. Again it feels tricky to do it without duplicating code from namespace.

Reply:
I approved, still a small comment.

A List of tuples (like `result`) can be transformed to a tuple of lists (like `labels, iris`) using zip:

```
labels, iris = zip(*result)
```

Reply:
Okay not quite, the result is a tuple of tuples:
>>> a = [(1,2), (3, 4), (5,6)]
>>> tuple(zip(*a))
((1, 3, 5), (2, 4, 6))

Reply:
Thanks, I did not think of using it with more than two arguments. I will merge already nevertheless so that you do not have to review again.